build/features.py

```diff
@@ -942,29 +943,34 @@ def description(self):
         return "Optimization and Tuning"
 
     def enabled(self, build):
-        build.flags['optimize'] = util.get_flags(build.env, 'optimize', 1)
-        if int(build.flags['optimize']):
+        build.flags['optimize'] = SCons.ARGUMENTS.get('optimize', '')
+        if build.flags['optimize'] is not 'portable':
```

Comment:
This would mean that we only disable the feature with the keyword 'disable'. There is no warning raised if someone uses '0' or 'off' or 'false'. The pythonic way to solve this is to define a set of allowed optimize values and then check if the given value is in the set

```Python
from sets import Set
self.optimize_options = Set(['portable ', 'tuned', '9', '8'])
build.flags['optimize'] = SCons.ARGUMENTS.get('optimize', None)
return build.flags['optimize'] in optimize_options
```

Then we will return if the given option is valid. Otherwise optimization is turned off.